### ISSUE TITLE
affectsConfiguration doesn´t handle the case when an attribute gets removed

### ISSUE THREAD

**Issue by egamma** [Microsoft/vscode#50566]
I have the following code:
```
if (e.affectsConfiguration('npm.scriptExplorerAction')
```
The check properly returns true when the 'npm.scriptExplorerAction' attribute is changed. However, when the attribute is removed completely and the default value of the setting is different than what it was before, then the check does not return true as expected.

The setting.json and the default value of the setting is `open`.
```json  
{
    "npm.scriptExplorerAction": "run",
}
```

When removing the setting completely then the affectsConfiguration test of the event does not return true, but it should do so.

**Comment by sandy081** [comment]
@egamma Looks like it is working for me.

Here is my extension code

```typescript

'use strict';

import * as vscode from 'vscode';

export function activate(context: vscode.ExtensionContext) {

    vscode.workspace.onDidChangeConfiguration(e => {
        vscode.window.showInformationMessage(`'npm.scriptExplorerAction' is changed: ${e.affectsConfiguration('npm.scriptExplorerAction')}`);
    });
}

```

And whenever I change/reset the setting, I see the prompt informing setting is affected

![kapture 2018-05-30 at 12 12 53](https://user-images.githubusercontent.com/10746682/40714500-192a4e36-6403-11e8-9f5e-a762fc30c5b5.gif)
 
Am I missing any steps here?

**Comment by egamma** [comment]
@sandy081 sorry, I got tricked by the fact that I´ve tested this with a workspace setting and I´ve overlooked that I had also defined a user setting for npm.scriptExplorerAction. Therefore deleting the workspace setting had the effect that the user setting is applied which had the value ´run´ and was different from the default value. Once I´ve deleted/resetted the user setting in my tests it all works as expected.
